packages/npm-packages/ruby-wasm-wasi/test/js_from_rb.test.ts

```diff
@@ -121,6 +121,24 @@ describe("Manipulation of JS from Ruby", () => {
       expr: `JS.global[:Math].call(:min, JS.eval("return 1"), JS.eval("return 2"))`,
       result: 1,
     },
+    {
+      expr: `
+        function_to_call = JS.eval('return { a: (callback) => { callback(1) } }')
+        b = nil
+        function_to_call.call(:a, Proc.new { |a| b = a })
+        b
+      `,
+      result: 1
+    },
+    {
+      expr: `
+        function_to_call = JS.eval('return { a: (callback) => { callback(1) } }')
+        b = nil
+        function_to_call.call(:a) { |a| b = a }
+        b
+      `,
+      result: 1
+    }
```

Comment:
Could you add a test case to check proc escaping works correctly by holding a callback and call it after callback registration?

Reply:
I'm not sure if I understand the test case correctly. Do you mean keeping a reference to the callback function in JS and calling it with another function? Something like:

```js
{
    expr: `
      function_to_call = JS.eval('let callback; return { a: (c) => { callback = c }, b: () => { callback(1) } }')
      b = nil
      function_to_call.call(:a) { |a| b = a }
      function_to_call.call(:b)
      b
    `,
    result: 1
}
```

Reply:
Yes, I mean something like it.